lib/PreviewProvider.js

```diff
@@ -277,15 +221,27 @@ PreviewProvider.prototype = {
   }),
 
   /**
+   * Find the metadata for each link in the database
+   */
+  _asyncFindItemsInDB: Task.async(function*(links) {
+    let cacheKeyArray = [];
+    links.forEach(link => cacheKeyArray.push(link.cache_key));
```

Comment:
I think you can just say

const cacheKeyArray = links.map(link => link.cache_key);